sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTraceExporterEventSource.cs

```diff
@@ -11,41 +12,64 @@ internal sealed class AzureMonitorTraceExporterEventSource : EventSource
     {
         private const string EventSourceName = "OpenTelemetry-TraceExporter-AzureMonitor";
         public static AzureMonitorTraceExporterEventSource Log = new AzureMonitorTraceExporterEventSource();
-
-        [NonEvent]
-        public void FailedExport(Exception ex)
+        public readonly IReadOnlyDictionary<string, EventLevel> EventLevelMap = new Dictionary<string, EventLevel>
         {
-            if (this.IsEnabled(EventLevel.Error, EventKeywords.All))
-            {
-                this.FailedExport(ex.ToInvariantString());
-            }
-        }
+            [EventLevelSuffix.Critical] = EventLevel.Critical,
+            [EventLevelSuffix.Error] = EventLevel.Error,
+            [EventLevelSuffix.Warning] = EventLevel.Warning,
+            [EventLevelSuffix.Informational] = EventLevel.Informational,
+            [EventLevelSuffix.Verbose] = EventLevel.Verbose
+        };
 
         [NonEvent]
-        public void SdkVersionCreateFailed(Exception ex)
+        public void Write(string name, object value)
```

Comment:
For context, this is being used like so:
`AzureMonitorTraceExporterEventSource.Log.Write($"FailedToExport{EventLevelSuffix.Error}", ex.LogAsyncException());`

This is a cool idea, but I'm not a fan of this approach.
Mainly because this requires string parsing, manipulation inside the EventSource to determine the EventLevel detail.
The event level detail was known at compile time and I think this step can be avoided.
Plus, this assumes that the period ('.') won't be used as punctuation in the message itself.

I would propose either, 
- call the correct method directly
- or pass an enum into this method and switch on the enum. (this avoids the string parsing).

Reply:
This is an inspiration from `DiagnosticSource` class, `StartActivity` and `StopActivity`
Ref: https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceActivity.cs#L30

I'm using a very simple string parser in `EventSource`, where execution should be fast with no data allocation.

Reply:
Additionally, because of `LastIndexOf('.')` this method will break the first time someone tries to write one or more sentences into this error message (using a period as punctuation). This limitation isn't referenced in the <summary> for this method. :(

Reply:
I merged the changes, will create a new PR to address it.